apps/emails-and-messages/src/lib/metadata-manager.ts

```diff
@@ -43,27 +44,14 @@ export async function fetchAllMetadata(client: Client): Promise<MetadataEntry[]>
   return data?.app?.privateMetadata.map((md) => ({ key: md.key, value: md.value })) || [];
 }
 
-export async function mutateMetadata(client: Client, metadata: MetadataEntry[]) {
-  // to update the metadata, ID is required
-  const { error: idQueryError, data: idQueryData } = await client
-    .query(FetchAppDetailsDocument, {})
-    .toPromise();
-
-  if (idQueryError) {
-    throw new Error(
-      "Could not fetch the app id. Please check if auth data for the client are valid."
-    );
-  }
-
-  const appId = idQueryData?.app?.id;
-
-  if (!appId) {
-    throw new Error("Could not fetch the app ID");
-  }
-
+export async function mutateMetadata(
+  client: Client,
+  metadata: MetadataEntry[],
+  authData: AuthData
```

Comment:
dont require AuthData if you need only appId. Kind of first sentence here 
https://en.wikipedia.org/wiki/Interface_segregation_principle